volume-providers/vpc/provider/provider.go

```diff
@@ -78,14 +78,13 @@ func NewProvider(conf *config.Config, logger *zap.Logger) (local.Provider, error
 	}
 
 	//Do config validation and enable only one generationType (i.e VPC-Classic | VPC-NG)
-	gcConfigFound := (conf.VPC.EndpointURL != "") && (conf.VPC.TokenExchangeURL != "") && (conf.VPC.APIKey != "") && (conf.VPC.ResourceGroupID != "")
-	g2ConfigFound := (conf.VPC.G2EndpointURL != "") && (conf.VPC.G2TokenExchangeURL != "") && (conf.VPC.G2APIKey != "") && (conf.VPC.G2ResourceGroupID != "")
+	gcConfigFound := (conf.VPC.EndpointURL != "" || conf.VPC.PrivateEndpointURL != "") && (conf.VPC.TokenExchangeURL != "" || conf.VPC.IKSTokenExchangePrivateURL != "") && (conf.VPC.APIKey != "") && (conf.VPC.ResourceGroupID != "")
+	g2ConfigFound := (conf.VPC.G2EndpointPrivateURL != "") && (conf.VPC.IKSTokenExchangePrivateURL != "") && (conf.VPC.G2APIKey != "") && (conf.VPC.G2ResourceGroupID != "")
```

Comment:
Do we need to check `conf.VPC.G2EndpointURL != "" || conf.VPC.G2EndpointPrivateURL != ""` ?
You are checking this in gcConfigFound

Reply:
added